modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -3980,11 +3980,18 @@ all. It returns the rotated rectangle in which the ellipse is inscribed. The fir
 is used. Developer should keep in mind that it is possible that the returned
 ellipse/rotatedRect data contains negative indices, due to the data points being close to the
 border of the containing Mat element.
+If the flag direct is true a direct ellipse fit algorithm is used decribed by @cite Fitzgibbon99
 
 @param points Input 2D point set, stored in std::vector\<\> or Mat
  */
 CV_EXPORTS_W RotatedRect fitEllipse( InputArray points );
 
+/** @overload
+@param points Input 2D point set, stored in std::vector\<\> or Mat
+@param direct Flag to use a direct ellipse fit algorithm
+*/
+CV_EXPORTS_AS(fitEllipse2) RotatedRect fitEllipse( InputArray points, bool direct );
```

Comment:
Thank you very much for your comments. I had already implemented the WSCG98 method and it did not work, but after I removed scaling it works now. Actually my code is a construction area. So give a few days to clean everything up.